src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// Taken from https://github.com/dotnet/cli/blob/8f7b58dd665f60300221a4eb5910a691749396aa/src/Microsoft.DotNet.MSBuildSdkResolver/Interop.NETStandard.cs
+
+#if NETSTANDARD2_0
+using System.Diagnostics;
+
+namespace Microsoft.Build.Locator
+{
+
+    using System;
+    using System.IO;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    internal static class DotNetSdkLocationHelper
+    {
+        public static VisualStudioInstance GetInstance(string workingDir)
+        {
+            string dotNetSdkPath = hostfxr_resolve_sdk(workingDir);
+
+            if (String.IsNullOrWhiteSpace(dotNetSdkPath))
+            {
+                return null;
+            }
+
+            if (!File.Exists(Path.Combine(dotNetSdkPath, "Microsoft.Build.dll")))
+            {
+                return null;
+            }
+
+            FileInfo dotnetAssemblyPath = new FileInfo(Path.Combine(dotNetSdkPath, "dotnet.dll"));
+
+            if (!dotnetAssemblyPath.Exists)
```

Comment:
For my education, what's the reason for using `File.Exists` above but `FileInfo.Exists` here?

Reply:
I decided to use a `FileInfo` because I'm checking its existence and then using the full path.  If I'm just checking the existence then I use `File.Exists()` generally.